bot/menu.py

```diff
@@ -49,8 +49,22 @@ async def callback(self, interaction: Interaction):
             The action implemented by the user that needs to be notified.
             In the context of the bot, the action is a slash command
         """
+        prepend_message_string = ""
+        if self.is_adding_subscriber:
+            prepend_message_string = "You are now subscribed to "
+        else:
+            prepend_message_string = "You are now unsubscribed from "
+        converted_menu_values = list(map(int, self.values))
+        semantic_selected_tags = [
+            option.emoji.name + " " + option.label if option.emoji else option.label
+            for option in self.options
+            if option.value in converted_menu_values

```

Comment:
this is really hard for me to understand whats going on, should we split this up so it a little easier to skim over in regards to how the logic is written out?

we have a mix of standard blocks with inline statements and its a little tough switching context to understand what we're trying to do.

Thoughts?

Reply:
I see where you are coming from. The original is a bit convoluted. I revised for clarity in the most recent commit. Same functionality is seen through testing.